src/libxrpl/protocol/SField.cpp

```diff
@@ -49,386 +49,32 @@ TypedField<T>::TypedField(private_access_tag_t pat, Args&&... args)
 // database:
 
 // Use macros for most SField construction to enforce naming conventions.
-#pragma push_macro("CONSTRUCT_UNTYPED_SFIELD")
-#undef CONSTRUCT_UNTYPED_SFIELD
+#pragma push_macro("UNTYPED_SFIELD")
+#undef UNTYPED_SFIELD
 
-// It would be possible to design the macros so that sfName and txtName would
-// be constructed from a single macro parameter.  We chose not to take that
-// path because then you cannot grep for the exact SField name and find
-// where it is constructed.  These macros allow that grep to succeed.
-#define CONSTRUCT_UNTYPED_SFIELD(sfName, txtName, stiSuffix, fieldValue, ...) \
-    SField const sfName(                                                      \
-        access, STI_##stiSuffix, fieldValue, txtName, ##__VA_ARGS__);         \
-    static_assert(                                                            \
-        std::string_view(#sfName) == "sf" txtName,                            \
-        "Declaration of SField does not match its text name")
+#define UNTYPED_SFIELD(name, stiSuffix, fieldValue, ...) \
+    SField const sf##name(                               \
+        access, STI_##stiSuffix, fieldValue, #name, ##__VA_ARGS__);
 
-#pragma push_macro("CONSTRUCT_TYPED_SFIELD")
-#undef CONSTRUCT_TYPED_SFIELD
+#pragma push_macro("TYPED_SFIELD")
+#undef TYPED_SFIELD
 
-#define CONSTRUCT_TYPED_SFIELD(sfName, txtName, stiSuffix, fieldValue, ...) \
-    SF_##stiSuffix const sfName(                                            \
-        access, STI_##stiSuffix, fieldValue, txtName, ##__VA_ARGS__);       \
-    static_assert(                                                          \
-        std::string_view(#sfName) == "sf" txtName,                          \
-        "Declaration of SField does not match its text name")
-
-// clang-format off
+#define TYPED_SFIELD(name, stiSuffix, fieldValue, ...) \
+    SF_##stiSuffix const sf##name(                     \
+        access, STI_##stiSuffix, fieldValue, #name, ##__VA_ARGS__);
```

Comment:
For consistency, please follow the same style as in Feature.cpp (order of operations where `#push_macro` and `#undef` go interleaved, then all new `#define`, then `#undef` and `#pop_macro` go interleaved)